hw_steering_src/dr_definer.py

```diff
@@ -0,0 +1,245 @@
+    "ib_l2": [[('sx_sniffer', 1), ('force_lb', 1), ('functional_lb', 1), ('reserved_at_3', 3), ('port_number', 2), ('sl', 4), ('qp_type', 2), ('lnh', 2), ('dlid', 16)], [('vl', 4), ('lrh_packet_length', 12), ('slid', 16)]],
+    "eth_l2_src": [[('smac_47_16', 32)], [('smac_15_0', 16), ('loopback_syndrome', 8), ('l3_type', 2), ('l4_type_bwc', 2), ('first_vlan_qualifier', 2), ('ip_fragmented', 1), ('functional_lb', 1)]],
+    "ipv4_src_dst": [[('source_address', 32)], [('destination_address', 32)]],
+    "ipv6_addr": [[('ipv6_address_127_96', 32)], [('ipv6_address_95_64', 32)], [('ipv6_address_63_32', 32)], [('ipv6_address_31_0', 32)]],
+    "flex_parser": [[('version', 3), ('proto_type', 1), ('reserved1', 1), ('ext_hdr_flag', 1), ('seq_num_flag', 1), ('pdu_flag', 1), ('msg_type', 8), ('msg_len', 8), ('teid', 32), ('seq_num', 16), ('pdu_num', 8), ('next_ext_hdr_type', 8), ('len', 8)]],
+    "oks2": [[('reserved_at_0', 10), ('second_mpls_ok', 1), ('second_mpls4_s_bit', 1), ('second_mpls4_qualifier', 1), ('second_mpls3_s_bit', 1), ('second_mpls3_qualifier', 1), ('second_mpls2_s_bit', 1), ('second_mpls2_qualifier', 1), ('second_mpls1_s_bit', 1), ('second_mpls1_qualifier', 1), ('second_mpls0_s_bit', 1), ('second_mpls0_qualifier', 1), ('first_mpls_ok', 1), ('first_mpls4_s_bit', 1), ('first_mpls4_qualifier', 1), ('first_mpls3_s_bit', 1), ('first_mpls3_qualifier', 1), ('first_mpls2_s_bit', 1), ('first_mpls2_qualifier', 1), ('first_mpls1_s_bit', 1), ('first_mpls1_qualifier', 1), ('first_mpls0_s_bit', 1), ('first_mpls0_qualifier', 1)]],
+    "oks1": [[('second_ipv4_checksum_ok', 1), ('second_l4_checksum_ok', 1), ('first_ipv4_checksum_ok', 1), ('first_l4_checksum_ok', 1), ('second_l3_ok', 1), ('second_l4_ok', 1), ('first_l3_ok', 1), ('first_l4_ok', 1), ('flex_parser7_steering_ok', 1), ('flex_parser6_steering_ok', 1), ('flex_parser5_steering_ok', 1), ('flex_parser4_steering_ok', 1), ('flex_parser3_steering_ok', 1), ('flex_parser2_steering_ok', 1), ('flex_parser1_steering_ok', 1), ('flex_parser0_steering_ok', 1), ('second_ipv6_extension_header_vld', 1), ('first_ipv6_extension_header_vld', 1), ('l3_tunneling_ok', 1), ('l2_tunneling_ok', 1), ('second_tcp_ok', 1), ('second_udp_ok', 1), ('second_ipv4_ok', 1), ('second_ipv6_ok', 1), ('second_l2_ok', 1), ('vxlan_ok', 1), ('gre_ok', 1), ('first_tcp_ok', 1), ('first_udp_ok', 1), ('first_ipv4_ok', 1), ('first_ipv6_ok', 1), ('first_l2_ok', 1)]],
+    "src_qp_gvmi": [[('loopback_syndrome', 8), ('l3_type', 2), ('l4_type_bwc', 2), ('first_vlan_qualifier', 2), ('reserved_at_e', 1), ('functional_lb', 1), ('source_gvmi', 16)], [('force_lb', 1), ('ip_fragmented', 1), ('source_is_requestor', 1), ('reserved_at_23', 5), ('source_qp', 24)]],
+    "voq": [[('reserved_at_0', 24), ('ecn_ok', 1), ('congestion', 1), ('profile', 2), ('internal_prio', 4)]],
+    "NOT_DEFINED": [],
+    }
```

Comment:
lets have new file "definer_parsers" and use for each definer a parser function. like:
mlx5_definer_parse_ib_l4
mlx5_definer_parse_ipv4_src_dst
...

Reply:
done